src/backend/kms/surface/mod.rs

```diff
@@ -1454,256 +1308,26 @@ impl SurfaceThreadState {
-                                            tracing::warn!(?err, "Failed to screencopy");
-                                            session
-                                                .user_data()
-                                                .get::<SessionData>()
-                                                .unwrap()
-                                                .lock()
-                                                .unwrap()
-                                                .reset();
-                                            frame.fail(FailureReason::Unknown);
-                                            continue;
-                                        }
-                                    };
-                                };
-                            }
-
-                            let transform = self.output.current_transform();
-
-                            match submit_buffer(
-                                frame,
+                            if let Err(err) = send_screencopy_result(
                                 &mut renderer,
-                                shm_buffer.then_some(fb.as_mut().unwrap()),
-                                transform,
-                                damage.as_deref(),
-                                sync,
+                                &self.output,
+                                &mut pre_postprocess_data,
+                                &tx,
+                                &frame_result,
+                                &elements,
+                                (&session, frame, res),
+                                now.into(),
                             ) {
-                                Ok(Some((frame, damage))) => {
-                                    if frame_result.is_empty {
-                                        frame.success(transform, damage, self.clock.now());
-                                    } else {
-                                        let _ = tx.send((frame, damage));
-                                    }
-                                }
-                                Ok(None) => {}
-                                Err(err) => {
-                                    session
-                                        .user_data()
-                                        .get::<SessionData>()
-                                        .unwrap()
-                                        .lock()
-                                        .unwrap()
-                                        .reset();
-                                    tracing::warn!(?err, "Failed to screencopy");
-                                }
+                                session
+                                    .user_data()
+                                    .get::<SessionData>()
+                                    .unwrap()
+                                    .lock()
+                                    .unwrap()
+                                    .reset();
```

Comment:
So we are not doing `frame.fail` here anymore. Is there a reason for that? Do we just re-try the frame next time?

Reply:
Agreed